packages/abi/src/matchers/sway-type-matchers.ts

```diff
@@ -0,0 +1,98 @@
+type SwayType =
+  | 'empty'
+  | 'bool'
+  | 'u8'
+  | 'u16'
+  | 'u32'
+  | 'u64'
+  | 'u256'
+  | 'b256'
+  | 'generic'
+  | 'string'
+  | 'stdString'
+  | 'option'
+  | 'result'
+  | 'enum'
+  | 'struct'
+  | 'b512'
+  | 'bytes'
+  | 'vector'
+  | 'tuple'
+  | 'array'
+  | 'assetId'
+  | 'evmAddress'
+  | 'rawUntypedPtr' // might not need it
+  | 'rawUntypedSlice'; // might not need it
+
+type Matcher = (type: string) => boolean;
+
+const empty: Matcher = (type) => type === '()';
+const bool: Matcher = (type) => type === 'bool';
+const u8: Matcher = (type) => type === 'u8';
+const u16: Matcher = (type) => type === 'u16';
+const u32: Matcher = (type) => type === 'u32';
+const u64: Matcher = (type) => type === 'u64';
+const u256: Matcher = (type) => type === 'u256';
+const b256: Matcher = (type) => type === 'b256';
+
+export const genericRegEx = /^generic ([^\s]+)$/m;
+const generic: Matcher = (type) => genericRegEx.test(type);
+
+export const stringRegEx = /str\[(?<length>[0-9]+)\]/;
+const string: Matcher = (type) => stringRegEx.test(type);
+
+export const tupleRegEx = /^\((?<items>.+)\)$/m;
+const tuple: Matcher = (type) => tupleRegEx.test(type);
+
+export const arrayRegEx = /\[(?<item>[\w\s\\[\]]+);\s*(?<length>[0-9]+)\]/;
+const array: Matcher = (type) => arrayRegEx.test(type);
```

Comment:
IMO, all constants should be `ALL_CAPS` to enhance readability.

```suggestion
export const GENERIC_REGEX = /^generic ([^\s]+)$/m;
const generic: Matcher = (type) => GENERIC_REGEX.test(type);

export const STRING_REGEX = /str\[(?<length>[0-9]+)\]/;
const string: Matcher = (type) => STRING_REGEX.test(type);

export const TUPLE_REGEX = /^\((?<items>.+)\)$/m;
const tuple: Matcher = (type) => TUPLE_REGEX.test(type);

export const ARRAY_REGEX = /\[(?<item>[\w\s\\[\]]+);\s*(?<length>[0-9]+)\]/;
const array: Matcher = (type) => ARRAY_REGEX.test(type);
```